### PR TITLE
Change Docker image file and detect containers in build script

### PR DESCRIPTION
**Pull request by tqre**
Changed Docker image file to the more frequently updated archlinux/archlinux. I also added docker.io to front for better podman compatibility. Didn't test with podman yet fully.

Building packages in container failed because of systemd checks because the part in script build_and_install_all.sh that would have triggered to disable those checks didn't fire.

I've been looking at GitHub Actions to set up a CI pipeline, so I had to make these work first. So far so good with those:
https://github.com/tqre/selinux/actions/runs/406799585/workflow

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,17 @@
+# Build SELinux packages for Arch linux in Docker/Podman container
+
 # Usage:
-#    sudo docker build -t arch-selinux-build .
-#
-# or with Podman on Arch Linux in July 2020:
-#    podman --cgroup-manager=cgroupfs build -t arch-selinux-build .
+#    docker build -t arch-selinux-build .
```
Comment (fishilico) [inline Dockerfile:4]: No. Please keep `sudo` here or document "running from `root`", as running `docker` directly from a non-root user is not a good security practice (it enables a very simple privilege escalation using `docker run -v /:/host -ti archlinux bash`)

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+# or   
+#    sudo podman build -t arch-selinux-build .
```
Comment (fishilico) [inline Dockerfile:6]: Why do you need `sudo` to use `podman`? The main point of using `podman` is to be able to use containers from a non-privileged user.

File: Dockerfile [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-#    sudo docker run -v "$(pwd)/pkgs:/packages" --rm -ti arch-selinux-build
+# Once the container is built, you can get the packages in "pkgs" directory with:
+#    docker run -v "$(pwd)/pkgs:/packages" --rm -ti arch-selinux-build
 # or
 #    podman run -v "$(pwd)/pkgs:/packages" --rm -ti arch-selinux-build
 
-# Use official Arch Linux Docker image (https://hub.docker.com/_/archlinux)
-FROM archlinux:latest
+# Use official Arch Linux Docker image (https://hub.docker.com/r/archlinux/archlinux)
+FROM docker.io/archlinux/archlinux:latest
```
Comment (fishilico) [inline Dockerfile:14]: https://github.com/archlinux/archlinux-docker#arch-linux-docker-image is not clear at all about what is the official image, and both images were updated recently (and actually `archlinux` *is* more recent than `archlinux/archlinux`, now):
```console
$ sudo docker images | grep archlinux
archlinux                      latest              ddcfcc25b371        12 hours ago        585MB
archlinux/archlinux            latest              f1713b6d04c3        23 hours ago        472MB
```

Anyway, the `archlinux/archlinux` looks "more official" so I agree on switching to it, and :+1: with adding `docker.io/`.

File: build_and_install_all.sh [hunk 4]
```diff
@@ -62,9 +62,7 @@ build() {
     rm -f "./$1/"*.pkg.tar.xz "./$1/"*.pkg.tar.xz.sig
     rm -f "./$1/"*.pkg.tar.zst "./$1/"*.pkg.tar.zst.sig
     # When building in a container, systemd's tests fail because of default Seccomp filters
-    if [ "$1" = 'systemd-selinux' ] && \
-        (systemd-detect-virt -q --container || systemd-detect-virt -q --private-users ) && \
-        grep '^Seccomp:\s*2$' /proc/self/status > /dev/null
+    if [ "$1" = 'systemd-selinux' ] && grep '^Seccomp:\s*2$' /proc/self/status > /dev/null
```
Comment (fishilico) [inline build_and_install_all.sh:65]: It is sad, but it is known bug in systemd: https://github.com/systemd/systemd/issues/15393

As systemd's developers consider it to be a feature request and not a bug (despite "docker" being listed in the man page, https://www.freedesktop.org/software/systemd/man/systemd-detect-virt.html), I have to agree with this change.

**Review by fishilico — CHANGES_REQUESTED** [review]
The changes in the scripts looks good, but not the changes in the comments of the files.

I consider that `docker` should only be used from a `root`-like account, contrary to `podman` which can be used from a non-`root` user. If you do not agree, please provide documentation/pointers in favor of using `docker` and `sudo podman`.

**Comment by fishilico** [comment]
Anyway, thanks for working on GitHub Actions! I was actually thinking of using them to build packages and host them as build artifacts (or push them to another repository which could be directly used as `pacman` package repo).

If you have time to spend on this, it would be great :)

File: Dockerfile (hunk 1 above)
Comment (tqre) [reply Dockerfile]: You are correct here, and it seems I have misunderstood the best practices I have looked at. I was well aware of a user having access to docker has essentially root on a system.

Namely I misinterpreted the 'run containers as non-root' -practice, which actually refers to the user working inside the container, not the user running docker itself.

File: Dockerfile (hunk 3 above)
Comment (tqre) [reply Dockerfile:15]: Here is the official explanation of the differences with the images, I put this link into the comment section.
https://gitlab.archlinux.org/archlinux/archlinux-docker#pipeline

**Review by tqre — COMMENTED** [review]
Thanks for comments. Made the changes you requested and added clarifications on docker image's officiality.

File: Dockerfile (hunk 2 above)
Comment (tqre) [reply Dockerfile]: My podman configurations are not correct, sorry. Totally unnecessary modification from me.

**Comment by tqre** [comment]
I'm fortunate to have lots of time in my hands at the moment, which might change soon though... I have made a simple working GitHub Actions pipeline now, and once this PR gets merged, I'll make another PR to start things with.

**Review by fishilico — APPROVED** [review]

**Pull request merged**